commands/version-compare

```diff
@@ -6,17 +6,43 @@ requires_array_support 'mapfile'
 # https://github.com/bevry/version-compare/blob/76f92ccde39e05a4af147eac293fe6b6743c9b73/source/index.ts
 # MIT License - 2020+ Benjamin Lupton <b@lupton.cc> (https://balupton.com)
 
-mapfile -t c < <(echo-split '.' -- "${1:?"USAGE: version-compare <CURRENT> <OTHER>"}")
-mapfile -t o < <(echo-split '.' -- "${2:?"USAGE: version-compare <current> <OTHER>"}")
+# help
+# =====================================
+# Arguments
 
 # help
-if is-help "$@"; then
+function help() {
 	cat <<-EOF >/dev/stderr
 		ABOUT:
-		?????
+		    Compare versions, if <CURRENT> is less <OTHER>
+
+		USAGE:
+		   USAGE: version-compare [ --help | -h ] -- <CURRENT> <OTHER>
+
 	EOF
+	if test "$#" -ne 0; then
+		echo-style $'\n' --error="ERROR:" $'\n' --red="$(echo-lines -- "$@")" >/dev/stderr
+	fi
 	return 22 # Invalid argument
-fi
+}
+
+# process
+while test "$#" -ne 0; do
+	item="$1"
+	shift
+	case "$item" in
+	'--help' | '-h') help ;;
+	'--'*) help "An unrecognised flag was provided: $item" ;;
+	'--') break ;;
+	*) help "An unrecognised argument was provided: $item" ;;
+	esac
+done
+
+mapfile -t c < <(echo-split '.' -- "$1")
+mapfile -t o < <(echo-split '.' -- "$2")
```

Comment:
args should be handled in the arg processing, as they won't exist after that while loop